Configuration/NativeMethods.cs

```diff
@@ -64,7 +64,9 @@ private struct AssemblyInfo
         public static string GetAssemblyPath(string name)
         {
             if (name == null)
-                throw new ArgumentNullException("name");
+                throw new ArgumentNullException("name");
+              
+            if( Type.GetType( "Mono.Runtime" ) != null ) return null;
 
```

Comment:
Nice find!

Could you give a bit of context? The condition looks reversed to me.

This code is copied from here: https://github.com/Azure/azure-sdk-for-net/blob/9154888d1ddbfe84cac0d86492f4bd6b99ed06ff/src/Common/Configuration/NativeMethods.cs, is Windows-specific, hard to test, and I'm hesitant to modify it. For this reason, this check should possibly be done in `AzureApplicationSettings` which is the only class that uses NativeMethods.cs.

Thanks.